src/rendering/loadingscreen.cpp

```diff
@@ -0,0 +1,706 @@
+                            ll,
+                            ur
+                        ) : 
+                    false;
+
+                    if (logoOverlap || loadingOverlap || messageOverlap || barOverlap) {
+                        continue;
+                    }
+
+                    
+                    // Test against all other boxes
+                    bool overlap = false;
+                    for (const Item& j : _items) {
+                        overlap |= rectOverlaps(j.ll, j.ur, ll, ur);
+
+                        if (overlap) {
+                            break;
+                        }
+                    }
+
+                    if (!overlap) {
+                        break;
+                    }
+                }
+
+                item.ll = ll;
+                item.ur = ur;
+
+                item.hasLocation = true;
+
+#ifdef LOADINGSCREEN_DEBUGGING
+                item.exhaustedSearch = (i == MaxNumberLocationSamples);
+#endif // LOADINGSCREEN_DEBUGGING
+
+            }
+
+            glm::vec4 color = [status = item.status]() {
+                switch (status) {
+                    case ItemStatus::Started:
+                        return ItemStatusColorStarted;
+                    case ItemStatus::Initializing:
+                        return ItemStatusColorInitializing;
+                    case ItemStatus::Finished:
+                        return ItemStatusColorFinished;
+                    case ItemStatus::Failed:
+                        return ItemStatusColorFailed;
+                    default:
+                        return glm::vec4(1.f);
+                }
+            }();
+            
+            if (item.status == ItemStatus::Finished) {
+                auto t = std::chrono::duration_cast<std::chrono::milliseconds>(now - item.finishedTime);
+
+                color.a = 1.f - static_cast<float>(t.count()) / static_cast<float>(TTL.count());
+            }
+
+#ifdef LOADINGSCREEN_DEBUGGING
+            if (item.exhaustedSearch) {
+                color = glm::vec4(0.f, 1.f, 1.f, 1.f);
+            }
+#endif // LOADINGSCREEN_DEBUGGING
+
+            renderer.render(
+                *_itemFont,
+                item.ll,
+                color,
+                "%s",
+                item.name.c_str()
+            );
+        }
+
+        _items.erase(
+            std::remove_if(
+                _items.begin(),
+                _items.end(),
+                [now](const Item& i) {
+                    if (i.status == ItemStatus::Finished) {
+                        return i.finishedTime > now + TTL;
+                    }
+                    else {
+                        return false;
+                    }
+                }
+            ),
+            _items.end()
+        );
+
+    }
+
+    glEnable(GL_CULL_FACE);
+    glEnable(GL_DEPTH_TEST);
+
+    std::this_thread::sleep_for(RefreshRate);
+    OsEng.windowWrapper().swapBuffer();
```

Comment:
Any particular reason why SGCT's render loop is not used for this?
Is there a risk things may look strange in fisheye rendering for example?

Reply:
Yeah..the initizliation happens before going into the render loop